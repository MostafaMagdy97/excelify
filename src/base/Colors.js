const baseColors = {
  primary: '#0070E0',
  secondary: '#06BEE1',
  success: '#44AF69',
  warning: '#FCAB10',
  danger: '#F8333C',
  black: '#1b2733',
  grey: '#637282',
  light: '#f5f5f5',
  white: '#fff',
  transparent: 'transparent',
};

const COLORS = {
  BLUE: 'blue',
  RED: 'red',
  DANGER: 'danger',
  PAGE_BACKGROUND: 'pageBackground',
  SUCCESS: 'success',
  WARNING: 'warning',
  LINK: 'link',
  DISABLED: 'disabled',
  TEXT: 'text',
  HELP_TEXT: 'helpText',
  BORDER: 'border',
  HEADER: 'header',
  WHITE: 'white',
  SEPARATOR: 'separator',
  TRANSPARENT: 'transparent',
};

const COLORS_VALUES = {
  [COLORS.BLUE]: baseColors.primary,
  [COLORS.PRIMARY_RED]: baseColors.secondary,
  [COLORS.DANGER]: baseColors.danger,
  [COLORS.PAGE_BACKGROUND]: baseColors.light,
  [COLORS.SUCCESS]: baseColors.success,
  [COLORS.WARNING]: baseColors.warning,
  [COLORS.LINK]: baseColors.primary,
  [COLORS.DISABLED]: baseColors.grey,
  [COLORS.TEXT]: baseColors.black,
  [COLORS.HELP_TEXT]: baseColors.grey,
  [COLORS.BORDER]: baseColors.light,
  [COLORS.HEADER]: baseColors.primary,
  [COLORS.WHITE]: baseColors.white,
  [COLORS.SEPARATOR]: baseColors.grey,
};

export { COLORS, COLORS_VALUES };
export default COLORS;
