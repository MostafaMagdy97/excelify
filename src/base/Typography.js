export const FONT_TYPES = {
  HERO_TITLE: 'heroTitle',
  SUPER_TITLE: 'superTitle',
  TITLE: 'title',
  HEADING: 'heading',
  SUBHEADING: 'subheading',
  BODY: 'body',
  CAPTION: 'caption',
};

export const FONT_SIZES = {
  // Breakpoints: ['400px', '576px', '786px', '992px']
  [FONT_TYPES.HERO_TITLE]: ['42px', '42px', '42x', '60px'],
  [FONT_TYPES.SUPER_TITLE]: ['32px', '32px', '32x', '30px'],
  [FONT_TYPES.TITLE]: ['21px', '21px', '20px', '20px'],
  [FONT_TYPES.HEADING]: ['19px', '19px', '18px', '18px'],
  [FONT_TYPES.SUBHEADING]: ['17px', '17px', '16px', '16px'],
  [FONT_TYPES.BODY]: ['15px', '15px', '14px', '14px'],
  [FONT_TYPES.CAPTION]: ['13px', '13px', '12px', '12px'],
};

export const FONT_WEIGHTS = {
  SEMI_BOLD: 600,
  NORMAL: 500,
  LIGHT: 400,
  SUPER_LIGHT: 300,
};

export const LINE_HEIGHTS = {
  [FONT_TYPES.HERO_TITLE]: '30px',
  [FONT_TYPES.SUPER_TITLE]: '30px',
  [FONT_TYPES.TITLE]: '24px',
  [FONT_TYPES.HEADING]: '24px',
  [FONT_TYPES.SUBHEADING]: '24px',
  [FONT_TYPES.BODY]: '24px',
  [FONT_TYPES.CAPTION]: '24px',
};
