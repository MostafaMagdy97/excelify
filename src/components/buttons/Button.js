import React, { Component } from 'react';
import PropTypes from 'prop-types';

import ButtonBase from './ButtonBase';
import Text from '../text/Text';
import Spinner from '../spinner/Spinner';
import { COLORS } from '../../base/Colors';

class Button extends Component {
  static propTypes = {
    children: PropTypes.string.isRequired,
    color: PropTypes.string,
    primary: PropTypes.bool,
    disabled: PropTypes.bool,
    isLoading: PropTypes.bool,
  };

  static defaultProps = {
    color: COLORS.BLUE,
    primary: true,
    disabled: false,
    isLoading: false,
  };

  state = {
    text: this.props.children,
  };

  /**
   * Decides button background and text color depends on the button type
   * and whether it's disabled or not
   * Types:
   * - Primary: Background will inherit color prop and the text will be white
   * - Secondary: Text will inherit color prop and the background will be transparent
   *
   * @returns {object} contains {textColor, buttonBackground}
   */
  getButtonColors = () => {
    const { color, primary, disabled, isLoading } = this.props;
    let textColor;
    let buttonBackground;

    if (primary) {
      textColor = disabled || isLoading ? COLORS.TEXT : COLORS.WHITE;
      buttonBackground = disabled || isLoading ? COLORS.DISABLED : color;
    } else {
      textColor = disabled || isLoading ? COLORS.TEXT : color;
      buttonBackground = COLORS.TRANSPARENT;
    }

    return {
      textColor,
      buttonBackground,
    };
  };

  render() {
    const { text } = this.state;
    const { isLoading, disabled } = this.props;
    const { textColor, buttonBackground } = this.getButtonColors();

    return (
      <ButtonBase {...this.props} disabled={disabled || isLoading} bg={buttonBackground}>
        {isLoading && <Spinner radius={16} color={textColor} mr={2} />}
        <Text tag="span" zIndex="1" width="max-content" color={textColor}>
          {text}
        </Text>
      </ButtonBase>
    );
  }
}

export default Button;
