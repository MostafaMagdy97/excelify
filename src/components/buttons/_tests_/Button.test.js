import React from 'react';
import { configure, render, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-enzyme';
import 'jest-styled-components';
import { ThemeProvider } from 'styled-components';

import Button from '../Button';
import theme from '../../../base/Theme';
import { COLORS } from '../../../base/Colors';

configure({ adapter: new Adapter() });

describe('Button', () => {
  it('should render', () => {
    const tree = render(
      <ThemeProvider theme={theme}>
        <Button>Button</Button>
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();

    const wrapper = shallow(<Button>Button</Button>);
    expect(wrapper.instance().getButtonColors()).toEqual({
      textColor: COLORS.WHITE,
      buttonBackground: COLORS.BLUE,
    });
  });

  it('should render color correctly with primary type', () => {
    const tree = render(
      <ThemeProvider theme={theme}>
        <Button color={COLORS.BLUE}>Button</Button>
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();

    const wrapper = shallow(<Button color={COLORS.PRIMARY_BLUE}>Button</Button>);
    expect(wrapper.instance().getButtonColors()).toEqual({
      textColor: COLORS.WHITE,
      buttonBackground: COLORS.BLUE,
    });
  });

  it('should render color correctly with outline type', () => {
    const tree = render(
      <ThemeProvider theme={theme}>
        <Button primary={false} color={COLORS.BLUE}>
          Button
        </Button>
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();

    const wrapper = shallow(
      <Button primary={false} color={COLORS.BLUE}>
        Button
      </Button>,
    );
    expect(wrapper.instance().getButtonColors()).toEqual({
      textColor: COLORS.BLUE,
      buttonBackground: COLORS.TRANSPARENT,
    });
  });

  it('should render disabled color correctly with primary type', () => {
    const tree = render(
      <ThemeProvider theme={theme}>
        <Button disabled color={COLORS.BLUE}>
          Button
        </Button>
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();

    const wrapper = shallow(
      <Button disabled color={COLORS.BLUE}>
        Button
      </Button>,
    );
    expect(wrapper.instance().getButtonColors()).toEqual({
      textColor: COLORS.TEXT,
      buttonBackground: COLORS.DISABLED_LIGHT,
    });
  });

  it('should render disabled color correctly with outline type', () => {
    const tree = render(
      <ThemeProvider theme={theme}>
        <Button disabled primary={false} color={COLORS.PRIMARY_BLUE}>
          Button
        </Button>
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();

    const wrapper = shallow(
      <Button disabled primary={false} color={COLORS.PRIMARY_BLUE}>
        Button
      </Button>,
    );
    expect(wrapper.instance().getButtonColors()).toEqual({
      textColor: COLORS.TEXT,
      buttonBackground: COLORS.TRANSPARENT,
    });
  });

  it('should render spinner when isLoading = true with primary type', () => {
    const tree = render(
      <ThemeProvider theme={theme}>
        <Button isLoading primary color={COLORS.PRIMARY_BLUE}>
          Button
        </Button>
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();

    const wrapper = shallow(
      <Button isLoading primary color={COLORS.PRIMARY_BLUE}>
        Button
      </Button>,
    );
    expect(wrapper.instance().getButtonColors()).toEqual({
      textColor: COLORS.TEXT,
      buttonBackground: COLORS.DISABLED_LIGHT,
    });
  });

  it('should render spinner when isLoading = true with outline type', () => {
    const tree = render(
      <ThemeProvider theme={theme}>
        <Button isLoading primary={false} color={COLORS.PRIMARY_BLUE}>
          Button
        </Button>
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();

    const wrapper = shallow(
      <Button isLoading primary={false} color={COLORS.PRIMARY_BLUE}>
        Button
      </Button>,
    );
    expect(wrapper.instance().getButtonColors()).toEqual({
      textColor: COLORS.TEXT,
      buttonBackground: COLORS.TRANSPARENT,
    });
  });

  it('should not trigger onClick when disabled', () => {
    let counter = 0;
    const count = () => {
      counter += 1;
    };

    const wrapper = mount(
      <Button disabled onClick={count}>
        Button
      </Button>,
    );
    wrapper.simulate('click');
    expect(counter).toEqual(0);
  });
});
