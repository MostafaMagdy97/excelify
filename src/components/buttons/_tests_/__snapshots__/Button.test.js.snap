// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: none;
  border-radius: 4px;
  height: 36px;
  padding: 0 32px;
  -webkit-transition: all 200ms ease-out;
  transition: all 200ms ease-out;
  outline: none;
  position: relative;
  min-width: 120px;
}

.c0:not(:disabled) {
  cursor: pointer;
}

.c1 {
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  color: #fff;
  font-weight: 500;
  font-size: 15px;
  line-height: 24px;
  z-index: 1;
  font-display: fallback;
}

@media screen and (min-width:400px) {
  .c1 {
    font-size: 15px;
  }
}

@media screen and (min-width:576px) {
  .c1 {
    font-size: 14px;
  }
}

@media screen and (min-width:786px) {
  .c1 {
    font-size: 14px;
  }
}

<button
  class="c0"
>
  <span
    class="c1"
    color="white"
    font-size="15px,15px,14px,14px"
    font-weight="500"
    lineheight="24px"
    tag="span"
    type="body"
    width="max-content"
    zindex="1"
  >
    Button
  </span>
</button>
`;

exports[`Button should render color correctly with outline type 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: none;
  border-radius: 4px;
  height: 36px;
  padding: 0 32px;
  -webkit-transition: all 200ms ease-out;
  transition: all 200ms ease-out;
  outline: none;
  position: relative;
  min-width: 120px;
  background-color: transparent;
}

.c0:not(:disabled) {
  cursor: pointer;
}

.c1 {
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  color: #1b2733;
  font-weight: 500;
  font-size: 15px;
  line-height: 24px;
  z-index: 1;
  font-display: fallback;
}

@media screen and (min-width:400px) {
  .c1 {
    font-size: 15px;
  }
}

@media screen and (min-width:576px) {
  .c1 {
    font-size: 14px;
  }
}

@media screen and (min-width:786px) {
  .c1 {
    font-size: 14px;
  }
}

<button
  class="c0"
>
  <span
    class="c1"
    color="text"
    font-size="15px,15px,14px,14px"
    font-weight="500"
    lineheight="24px"
    tag="span"
    type="body"
    width="max-content"
    zindex="1"
  >
    Button
  </span>
</button>
`;

exports[`Button should render color correctly with primary type 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: none;
  border-radius: 4px;
  height: 36px;
  padding: 0 32px;
  -webkit-transition: all 200ms ease-out;
  transition: all 200ms ease-out;
  outline: none;
  position: relative;
  min-width: 120px;
}

.c0:not(:disabled) {
  cursor: pointer;
}

.c1 {
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  color: #fff;
  font-weight: 500;
  font-size: 15px;
  line-height: 24px;
  z-index: 1;
  font-display: fallback;
}

@media screen and (min-width:400px) {
  .c1 {
    font-size: 15px;
  }
}

@media screen and (min-width:576px) {
  .c1 {
    font-size: 14px;
  }
}

@media screen and (min-width:786px) {
  .c1 {
    font-size: 14px;
  }
}

<button
  class="c0"
>
  <span
    class="c1"
    color="white"
    font-size="15px,15px,14px,14px"
    font-weight="500"
    lineheight="24px"
    tag="span"
    type="body"
    width="max-content"
    zindex="1"
  >
    Button
  </span>
</button>
`;

exports[`Button should render disabled color correctly with outline type 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: none;
  border-radius: 4px;
  height: 36px;
  padding: 0 32px;
  -webkit-transition: all 200ms ease-out;
  transition: all 200ms ease-out;
  outline: none;
  position: relative;
  min-width: 120px;
  background-color: transparent;
}

.c0:not(:disabled) {
  cursor: pointer;
}

.c1 {
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  color: #1b2733;
  font-weight: 500;
  font-size: 15px;
  line-height: 24px;
  z-index: 1;
  font-display: fallback;
}

@media screen and (min-width:400px) {
  .c1 {
    font-size: 15px;
  }
}

@media screen and (min-width:576px) {
  .c1 {
    font-size: 14px;
  }
}

@media screen and (min-width:786px) {
  .c1 {
    font-size: 14px;
  }
}

<button
  class="c0"
  disabled=""
>
  <span
    class="c1"
    color="text"
    font-size="15px,15px,14px,14px"
    font-weight="500"
    lineheight="24px"
    tag="span"
    type="body"
    width="max-content"
    zindex="1"
  >
    Button
  </span>
</button>
`;

exports[`Button should render disabled color correctly with primary type 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: none;
  border-radius: 4px;
  height: 36px;
  padding: 0 32px;
  -webkit-transition: all 200ms ease-out;
  transition: all 200ms ease-out;
  outline: none;
  position: relative;
  min-width: 120px;
  background-color: #637282;
}

.c0:not(:disabled) {
  cursor: pointer;
}

.c1 {
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  color: #1b2733;
  font-weight: 500;
  font-size: 15px;
  line-height: 24px;
  z-index: 1;
  font-display: fallback;
}

@media screen and (min-width:400px) {
  .c1 {
    font-size: 15px;
  }
}

@media screen and (min-width:576px) {
  .c1 {
    font-size: 14px;
  }
}

@media screen and (min-width:786px) {
  .c1 {
    font-size: 14px;
  }
}

<button
  class="c0"
  disabled=""
>
  <span
    class="c1"
    color="text"
    font-size="15px,15px,14px,14px"
    font-weight="500"
    lineheight="24px"
    tag="span"
    type="body"
    width="max-content"
    zindex="1"
  >
    Button
  </span>
</button>
`;

exports[`Button should render spinner when isLoading = true with outline type 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: none;
  border-radius: 4px;
  height: 36px;
  padding: 0 32px;
  -webkit-transition: all 200ms ease-out;
  transition: all 200ms ease-out;
  outline: none;
  position: relative;
  min-width: 120px;
  background-color: transparent;
}

.c0:not(:disabled) {
  cursor: pointer;
}

.c2 {
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  color: #1b2733;
  font-weight: 500;
  font-size: 15px;
  line-height: 24px;
  z-index: 1;
  font-display: fallback;
}

.c1 {
  -webkit-animation: cZxgpV 2s linear infinite;
  animation: cZxgpV 2s linear infinite;
  width: 16px;
  height: 16px;
  margin-right: 8px;
}

.c1 .path {
  stroke: #1b2733;
  stroke-linecap: round;
  -webkit-animation: kuSBtA 1.5s ease-in-out infinite;
  animation: kuSBtA 1.5s ease-in-out infinite;
}

@media screen and (min-width:400px) {
  .c2 {
    font-size: 15px;
  }
}

@media screen and (min-width:576px) {
  .c2 {
    font-size: 14px;
  }
}

@media screen and (min-width:786px) {
  .c2 {
    font-size: 14px;
  }
}

<button
  class="c0"
  disabled=""
>
  <svg
    class="c1"
    color="text"
    radius="16"
    viewBox="0 0 50 50"
  >
    <circle
      class="path"
      cx="25"
      cy="25"
      fill="none"
      r="20"
      stroke-width="4"
    >
      
    </circle>
  </svg>
  <span
    class="c2"
    color="text"
    font-size="15px,15px,14px,14px"
    font-weight="500"
    lineheight="24px"
    tag="span"
    type="body"
    width="max-content"
    zindex="1"
  >
    Button
  </span>
</button>
`;

exports[`Button should render spinner when isLoading = true with primary type 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: none;
  border-radius: 4px;
  height: 36px;
  padding: 0 32px;
  -webkit-transition: all 200ms ease-out;
  transition: all 200ms ease-out;
  outline: none;
  position: relative;
  min-width: 120px;
  background-color: #637282;
}

.c0:not(:disabled) {
  cursor: pointer;
}

.c2 {
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  color: #1b2733;
  font-weight: 500;
  font-size: 15px;
  line-height: 24px;
  z-index: 1;
  font-display: fallback;
}

.c1 {
  -webkit-animation: cZxgpV 2s linear infinite;
  animation: cZxgpV 2s linear infinite;
  width: 16px;
  height: 16px;
  margin-right: 8px;
}

.c1 .path {
  stroke: #1b2733;
  stroke-linecap: round;
  -webkit-animation: kuSBtA 1.5s ease-in-out infinite;
  animation: kuSBtA 1.5s ease-in-out infinite;
}

@media screen and (min-width:400px) {
  .c2 {
    font-size: 15px;
  }
}

@media screen and (min-width:576px) {
  .c2 {
    font-size: 14px;
  }
}

@media screen and (min-width:786px) {
  .c2 {
    font-size: 14px;
  }
}

<button
  class="c0"
  disabled=""
>
  <svg
    class="c1"
    color="text"
    radius="16"
    viewBox="0 0 50 50"
  >
    <circle
      class="path"
      cx="25"
      cy="25"
      fill="none"
      r="20"
      stroke-width="4"
    >
      
    </circle>
  </svg>
  <span
    class="c2"
    color="text"
    font-size="15px,15px,14px,14px"
    font-weight="500"
    lineheight="24px"
    tag="span"
    type="body"
    width="max-content"
    zindex="1"
  >
    Button
  </span>
</button>
`;
